db/schema.go

```diff
@@ -75,62 +66,110 @@ type Policy struct {
 // }
 // }
 
-// A generic element of a policy, specified as a `attribute`, `operator` and `value`
-// Attribute is formatted as a path, i.e location.city, retrieval.success_rate
-// Some examples:
+type ComparisonOperator string
+
+const (
+	GreaterThan        ComparisonOperator = ">"
+	LessThan           ComparisonOperator = "<"
+	EqualTo            ComparisonOperator = "="
+	GreaterThanOrEqual ComparisonOperator = ">="
+	LessThanOrEqual    ComparisonOperator = "<="
+	IncludedIn         ComparisonOperator = "in"
+	ExcludedFrom       ComparisonOperator = "nin"
+	NotEqualTo         ComparisonOperator = "!="
+)
+
+func (s *ComparisonOperator) Scan(value interface{}) error {
+	strVal, ok := value.(string)
+	if !ok {
+		return errors.New("failed to scan TenantSpState")
+	}
+	*s = ComparisonOperator(strVal)
+	return nil
+}
+
+func (s *ComparisonOperator) Value() (driver.Value, error) {
+	return string(*s), nil
+}
+
+// Generic SP <-> Tenant Eligibility Clause, specified as a `attribute`, `operator` and `value`
+// examples:
 // location.country ComparisonOperator.IncludedIn [CAN, USA]
 // retrieval.success_rate ComparisonOperator.GreaterThan 0.98
-type Clause struct {
-	gorm.Model
-	PolicyID        uint               `json:"policy_id"`
-	ClauseAttribute string             `json:"attribute"`
-	ClauseOperator  ComparisonOperator `json:"operator"`
-	ClauseValue     string             `json:"value"`
+type TenantSPEligibilityClauses struct {
+	ModelBase
+	TenantID        ID                 `json:"tenant_id"`
+	ClauseAttribute string             `json:"attribute" gorm:"not null"`
+	ClauseOperator  ComparisonOperator `json:"operator" gorm:"type:comparison_operator;not null"`
```

Comment:
`type:comparison_operator` matches the `enum` manually entered in the `migrations.go` file